kotest-framework/kotest-framework-multiplatform-plugin-gradle/src/test/kotlin/io/kotest/framework/multiplatform/gradle/KotestPluginExtensionSpec.kt

```diff
@@ -0,0 +1,198 @@
+package io.kotest.framework.multiplatform.gradle
+
+import io.kotest.assertions.asClue
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.spec.style.scopes.FunSpecContainerScope
+import io.kotest.framework.multiplatform.gradle.util.GradleGroovyProjectTest.Companion.gradleGroovyProjectTest
+import io.kotest.framework.multiplatform.gradle.util.GradleKtsProjectTest.Companion.gradleKtsProjectTest
+import io.kotest.framework.multiplatform.gradle.util.GradleProjectTest
+import io.kotest.inspectors.forOne
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+import io.kotest.matchers.string.shouldNotContain
+import org.gradle.testkit.runner.TaskOutcome
+
+class KotestPluginExtensionSpec : FunSpec({
+
+   context("verify Kotest compiler version can be set manually") {
+
+      context("kts") {
+         val gradleTest = gradleKtsProjectTest {
+            buildGradleKts = """
+plugins {
+  base
+  id("io.kotest.multiplatform")
+  kotlin("multiplatform")
+}
+
+kotest {
+  kotestCompilerPluginVersion.set("1.2.3")
+}
+
+kotlin {
+  jvm()
+
+  sourceSets {
+    val commonTest by getting {
+      dependencies {
+        // the version here is not considered
+        implementation("io.kotest:kotest-framework-engine:4.5.6")
+      }
+    }
+  }
+}
+
+val printKotestCompilerPluginVersion by tasks.registering {
+  val kotestCompilerPluginVersion = kotest.kotestCompilerPluginVersion
+  doLast {
+    logger.lifecycle("Kotest Compiler version: " + kotestCompilerPluginVersion.orNull)
+  }
+}
+""".trimIndent()
+         }
+         `verify Gradle can configure the project`(gradleTest)
+         `verify Kotest compiler version is set`(gradleTest, "1.2.3")
+         `verify Kotest plugin warnings`(gradleTest)
+      }
+
+      context("groovy") {
```

Comment:
Do we need to test both Kotlin and Groovy scripts? I wouldn't expect to find any issues with the Groovy test that wouldn't already be uncovered by the Kotlin test.

Reply:
Better safe than sorry 🙂  Plus the test setup is there, in case a Groovy test is needed in the future - it will be easier to copy-paste and adjust the existing test. But you're right, I also don't expect any issues.